mssw/Acme-Spinodal-Utils/lib/Acme/Spinodal/Utils.pm

```diff
@@ -0,0 +1,190 @@
+package Acme::Spinodal::Utils;
+
+use 5.006;
+use strict;
+use warnings;
+
+use Scalar::Util qw( looks_like_number );
+use Carp qw( croak );
+
+=head1 NAME
+
+Acme::Spinodal::Utils - Some utility functions for me, Spinodal
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+
+
+=head1 SYNOPSIS
+
+Provides some helpful utility functions
+
+=head1 EXPORT
+
+=head2 @EXPORT:
+-none-
+
+=head2 @EXPORT_OK
+sum
+
+=head2 %EXPORT_TAGS
+all (everything)
+
+
+=cut
+
+our @EXPORT = qw(
+);
+
+our @EXPORT_OK = qw(
+    sum
+);
+
+our %EXPORT_TAGS = (
+    all => [ @EXPORT, @EXPORT_OK],
+);
+
+=head1 SUBROUTINES/METHODS
+
+=head2 sum
+
+Adds a series of numbers together
+
+    my $result = sum( 1, 2.1, 3, ... );
+
+=cut
+
+sub sum { 
+    my $total;
+    my $num = shift // 0;
+    
+    use Data::Dumper;
+    $total = _check_number($num);
+    
+    while( (defined ($num = shift)) && (defined _check_number( $num )) ){
```

Comment:
This would be simpler as `foreach my $num ( map _check_number( $_ ), @_ ) { $total *= $num }`